src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -49,7 +52,9 @@ public PatchXpathFilterElement(String fileName, List<List<Integer>> lineRangeLis
 
     @Override
     public boolean accept(TreeWalkerAuditEvent event) {
-        return isFileNameMatching(event) && isLineMatching(event);
+        return isFileNameMatching(event)
+                && (isContextStrategyCheck(event)
```

Comment:
We should check if strategy='context' to start context suppressions. This will check context strategy even if we selected 'patchedline' strategy in unit test.

Reply:
done